src/app.ts

```diff
@@ -13,14 +17,102 @@ interface TemperatureSummary {
   average: number
 }
 
+class dataByTime implements TemperatureReading {
+  public dataTemperature: Array<number> = []
+  // I created a new Class by error not definition
+  public TSummary: ClassTSummary = new ClassTSummary(0)
+  constructor(
+    public time: Date,
+    public city: string,
+    public temperature: number,
+  ) {
+    this.dataTemperature.push(temperature)
+  }
+
+  addTmpe(num: number): void {
+    this.dataTemperature.push(num)
+  }
+}
+
+class ClassTSummary implements TemperatureSummary {
+  constructor(temperature: number) {
+    this.first = this.last = this.high = this.low = this.average = temperature
+  }
+  first
+  last
+  high
+  low
+  average
+}
```

Comment:
I don't think you need to use classes here and I think it actually makes this maybe a little more heavy than it needs to be? It also kind of made it a little harder for me to understand/read. Could we refactor a bit to not use classes?

Reply:
Okey, I am using Dictionary into Dictionary is very fast.